### ISSUE TITLE
DBT 0.17.0rc3 - Compilation failure regression

### ISSUE THREAD

**Issue by AcidFlow** [fishtown-analytics/dbt#2504]
### Describe the bug

When trying to compile a project having some schema tests using `dbt_utils.equality` the compilation fails because the model does not exists.
However calling DBT run works.

The compilation failure happens when using DBT 0.17.0rc3, however compiling the same project using DBT 0.16.1 succeeds.

### Steps To Reproduce

The following ZIP file includes a dummy project to reproduce the issue.
[dummy.zip](https://github.com/fishtown-analytics/dbt/files/4705434/dummy.zip)

You will need to configure a profile for it though.

Extract the project and run:
```
dbt deps
dbt compile
```

### Expected behavior

The compilation succeeds on both DBT 0.16.1 and 0.17.0rc3

### Actual behavior

The compilation succeeds for DBT 0.16.1 and fails for 0.17.0rc3 with the following error:

```
Encountered an error:
Runtime Error
  Database Error in test dbt_utils_equality_my_second_dbt_model_id__ref_my_first_dbt_model_ (models/example/schema.yml)
    002003 (42S02): SQL compilation error:
    Table 'DB.SCHEMA.MY_SECOND_DBT_MODEL' does not exist or not authorized.
```

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [ ] bigquery
- [x] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
dbt --version
installed version: 0.17.0-rc3
   latest version: 0.16.1

Your version of dbt is ahead of the latest release!

Plugins:
  - snowflake: 0.17.0rc3
  - postgres: 0.17.0rc3
  - redshift: 0.17.0rc3
  - bigquery: 0.17.0rc3
```

**The operating system you're using:**  Linux Mint 19.3

**The output of `python --version`:** Python 3.6.9

**Comment by beckjake** [comment]
Thank you for the writeup and the _wonderful_ reproduction case! 🙏  I believe the root cause here is that we switched snowflake's `get_columns_in_relation` to use `describe table`, which results in a database error -> an exception when the relation is missing. Other adapters select from the information_schema and just return an empty list when there are no columns. The new snowflake behavior seems like a regression to me for sure!

**Comment by beckjake** [comment]
Fixed in #2509
